File: cpp/bc2139.cpp
#include <iostream>

using namespace std;

int main()
{
	int dMo[] = {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
	int day, mo, sDay;
	string res;
	
	while(cin >> mo >> day)
	{
		sDay = 0;
		res = "";
		
		if(mo == 12)
		{
			switch(day)
			{
				case 25: res = "E natal!"; break;
				case 24: res = "E vespera de natal!"; break;
				default: res = "Ja passou!"; break;
			}
			cout << res << endl;
		}
		else
		{
			for(int i = mo; i < 12; i++)
				sDay += dMo[i-1];
				
			sDay -= day;
			sDay += 25;
		
			cout << "Faltam " << sDay << " dias para o natal!" << endl;
 		}
	}
}

File: python/bc2139.py
dMo = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

while True:
    sDay = 0
    res = ""
    try:
        cVal = input().split(" ")
        mo = int(cVal[0])
        day = int(cVal[1])
    except EOFError:
        break
    if mo == 12:
        if day == 25: res = "E natal!"
        elif day == 24: res = "E vespera de natal!"
        else: res = "Ja passou!"
    else:
        for c in range(mo, 12): sDay += dMo[c-1]

        sDay -= day
        sDay += 25

        res = "Faltam {} dias para o natal!".format(sDay)

    print(res)